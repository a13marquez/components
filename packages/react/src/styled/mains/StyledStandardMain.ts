import styled from 'styled-components';
import { StandardStyles } from '../shared-styles';

export const StyledStandardMain = styled.main`
  ${StandardStyles}
`