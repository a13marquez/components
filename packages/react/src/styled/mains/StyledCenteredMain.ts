import styled from 'styled-components';
import { StyledStandardMain } from './StyledStandardMain';

interface StyledCenteredMainProps {
  backgroundColor?: string;
}

export const StyledCenteredMain = 
  styled(StyledStandardMain)<StyledCenteredMainProps>`
  display: flex;
  position: fixed;
  top: 50%;
  left: 50%;
  overflow-x: hidden;
  overflow-y: auto;
  max-width: 1088px;
  max-height: 648px;
  width: calc(100% - 64px);
  height: calc(100% - 64px);
  border-radius: 4px;
  background-color: ${props => props.backgroundColor || '#fff'};
`