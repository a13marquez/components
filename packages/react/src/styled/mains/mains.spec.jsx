import React from 'react';
import { mount } from 'enzyme';
import 'jest-styled-components';

import StyledStandardMain from './StyledStandardMain';
import StyledCenteredMain from './StyledCenteredMain';

test('It renders Styled Standard Main', () => {
  const StandardMain = mount(<StyledStandardMain />);
  expect(StandardMain).toMatchSnapshot();
});

test('It renders Styled Main App', () => {
  const MainApp = mount(<StyledCenteredMain />);
  expect(MainApp).toMatchSnapshot();
});

test('Styled Standard Row have the correct styles', () => {
  const StandardMain = mount(<StyledStandardMain />);
  expect(StandardMain).toHaveStyleRule('margin', '0');
  expect(StandardMain).toHaveStyleRule('padding', '0');
  expect(StandardMain).toHaveStyleRule('border', '0');
  expect(StandardMain).toHaveStyleRule('vertical-align', 'baseline');
  expect(StandardMain).toHaveStyleRule('font', '100%');
});

test('Styled Product Title Row have the correct styles', () => {
  const MainApp = mount(<StyledCenteredMain />);
  expect(MainApp).toHaveStyleRule('margin', '0');
  expect(MainApp).toHaveStyleRule('padding', '0');
  expect(MainApp).toHaveStyleRule('border', '0');
  expect(MainApp).toHaveStyleRule('vertical-align', 'baseline');
  expect(MainApp).toHaveStyleRule('font', '100%');

  expect(MainApp).toHaveStyleRule('position', 'fixed');
  expect(MainApp).toHaveStyleRule('top', '50%');
  expect(MainApp).toHaveStyleRule('left', '50%');
  expect(MainApp).toHaveStyleRule('display', 'flex');
  expect(MainApp).toHaveStyleRule('overflow-x', 'hidden');
  expect(MainApp).toHaveStyleRule('overflow-y', 'auto');
  expect(MainApp).toHaveStyleRule('max-width', '1088px');
  expect(MainApp).toHaveStyleRule('max-height', '648px');
  expect(MainApp).toHaveStyleRule('width', 'calc(100% - 64px)');
  expect(MainApp).toHaveStyleRule('height', 'calc(100% - 64px)');
  expect(MainApp).toHaveStyleRule('border-radius', '4px');
  expect(MainApp).toHaveStyleRule('background-color', '#fff');
});
