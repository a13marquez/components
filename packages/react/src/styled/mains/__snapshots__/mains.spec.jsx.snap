// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It renders Styled Main App 1`] = `ReactWrapper {}`;

exports[`It renders Styled Standard Main 1`] = `ReactWrapper {}`;
